<launch>
  <arg name="gui" default="true" />
  <arg name="convert_image" default="false" />

  <node name="infratec_image_convert_mono" pkg="infratec_image_processing" type="irb_image_convert_mono.py" output="screen" if="$(arg convert_image)">
    <param name="thermal_image" type="str" value="/infratec/image_raw"/>
  </node>
  
  <!-- Publish: contours (opencv_apps/ContourArrayStamped) -Detected contours. Each contour is stored as a vector of point -->
  <!-- <include file="$(find opencv_apps)/launch/find_contours.launch">
    <arg name="image"  value="/infratec/image_converted_mono" />
  </include> -->

  <!-- Publish: image (sensor_msgs/Image) Ouptut edge image. -->
  <!-- <include file="$(find opencv_apps)/launch/edge_detection.launch">
    <arg name="image"  value="/infratec/image_converted_mono" />
  </include> -->

   <!-- Publish: hulls (opencv_apps/ContourArrayStamped) - Detected convex hulls of contours. Each contour is stored as a vector of point -->
  <!-- <include file="$(find opencv_apps)/launch/convex_hull.launch">
    <arg name="image"  value="/infratec/image_converted_mono" />
  </include> -->

  <include file="$(find opencv_apps)/launch/general_contours.launch">
    <arg name="image"  value="/infratec/image_meltpool_bin" />
    <arg name="debug_view" value="false"/>
  </include>
<!-- 
  <include file="$(find opencv_apps)/launch/fback_flow.launch">
    <arg name="image"  value="/infratec/image_converted_mono" />
  </include> -->

  <!-- <include file="$(find opencv_apps)/launch/simple_flow.launch">
    <arg name="image"  value="/infratec/image_converted_mono" />
  </include> -->


  <!-- <include file="$(find opencv_apps)/launch/lk_flow.launch">
    <arg name="image"  value="/infratec/image_converted_mono" />
  </include> -->



  <!-- <node name="plot_nit" pkg="rqt_plot" type="rqt_plot" args="/usb_cam/geometry/minor_axis /usb_cam/moments/minor_axis" if="$(arg gui)" /> -->

</launch>
